cql3/restrictions/restriction.hh

```diff
@@ -41,19 +41,30 @@
 
 #pragma once
 
+#include <numeric>
+#include <optional>
+#include <sstream>
 #include <variant>
 #include <vector>
 
+#include <fmt/ostream.h>
```

Comment:
Will do, but there will still be a `#include <fmt/core.h>` here, because I'd like to declare the `formatter<expression>` specialization (requested [here](https://github.com/scylladb/scylla/pull/5763#discussion_r435102158)) in this header file.  That way any user of `expression` can invoke `fmt::join()` on `expression` collections.

I hope `#include <fmt/core.h>` is acceptable.  That header currently `#include`s only standard headers:
https://sourcegraph.com/github.com/fmtlib/fmt/-/blob/include/fmt/core.h?subtree=true#L11-L18